src/server/main.cpp

```diff
@@ -21,13 +30,67 @@ const int SIZE = 2;
 
 uint16_t *shared_mem;
 
+void doUpdate(const SwtFB &fb, const swtfb_update &s) {
+  auto mxcfb_update = s.update;
+  auto rect = mxcfb_update.update_region;
+
+#ifdef DEBUG_DIRTY
+  std::cerr << "Dirty Region: " << rect.left << " " << rect.top << " "
+            << rect.width << " " << rect.height << endl;
+#endif
+
+  // For now, we are using two waveform modes:
+  // 0: init (same as GL16)
+  // 1: DU - direct update, fast
+  // 2: GC16 - high fidelity (slow)
+  // 3: GL16 - what the rm is using
+  // 8: highlight (same as high fidelity)
+
+  int waveform = mxcfb_update.waveform_mode;
+  // full = 1, partial = 0
+  auto update_mode = mxcfb_update.update_mode;
+
```

Comment:
it looks like the translation from old waveform -> new waveform was removed. (there was some logic from bokluk here to add / change flags). are we sure its not needed anymore? (what was circumstances that required it)

Reply:
In xochitl they actual still have the old method and it maps to the new one by just doing `updatemode = old_mode & 1` and `if (sync) { updateMode |= 2; }` So I removed the old broken stuff

Reply:
awesome! :-D